dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -668,4 +688,207 @@ private SourceCodeRepoInterface getSourceCodeRepoInterface(String gitUrl, User u
         Helper.checkEntry(result);
         return result.getVersions();
     }
+
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/{workflowId}/dag/{workflowVersionId}")
+    @ApiOperation(value = "Get the DAG for a given workflow version", notes = "", response = String.class)
+    public String getWorkflowDag(@ApiParam(value = "workflowId", required = true) @PathParam("workflowId") Long workflowId,
+            @ApiParam(value = "workflowVersionId", required = true) @PathParam("workflowVersionId") Long workflowVersionId)  {
+        // TODO : Make this function modular (break into separate functions for getting nodes from WDL, CWL and turning into DAG)
+        // This will make it easier in the future to add new types (also better coding practice)
+        Workflow workflow = workflowDAO.findById(workflowId);
+        Set<WorkflowVersion> workflowVersions = workflow.getVersions();
+        WorkflowVersion workflowVersion = null;
+
+        for (WorkflowVersion wv : workflowVersions) {
+            if (wv.getId() == workflowVersionId) {
+                workflowVersion = wv;
+                break;
+            }
+        }
+
+        // Find main descriptor
+        SourceFile mainDescriptor = null;
+        for (SourceFile sourceFile : workflowVersion.getSourceFiles()) {
+            if (sourceFile.getPath().equals(workflowVersion.getWorkflowPath())) {
+                mainDescriptor = sourceFile;
+                break;
+            }
+        }
+
+        if (mainDescriptor != null) {
+            File tmpDir = Files.createTempDir();
+            File tempMainDescriptor = null;
+            try {
+                // Write main descriptor to file
+                tempMainDescriptor = File.createTempFile("main", "descriptor", tmpDir);
```

Comment:
Add a comment/GitHub issue here, the use of temporary files is not-needed here and introduces possible problems.